lang/java.md

```diff
@@ -0,0 +1,114 @@
+
+✦ Wide range of Mobile java Applications
+
+✦ Scientific Computing Applications
+
+✦ Use for Big Data Analytics
+
+✦ Java Programming of Hardware devices
+
+✦ Used for Server-Side Technologies like [Apache](https://commons.apache.org/), [JBoss](https://developers.redhat.com/products/eap/overview), [GlassFish](https://javaee.github.io/glassfish/), etc.
+
+    
+## Why to use Java?
+Java was designed to be easy to use and is therefore easy to write, compile, debug, and learn than other programming languages. Java is object-oriented. This allows you to 
+create modular programs and reusable code. Java is platform-independent.
+
+Java is a solid core language. Web and non-web environments, all operating systems, all types of devices you’ll find Java everywhere. For those in favor of learning Java 
+first, a key factor is that it teaches you to think like a programmer. You’ll learn the language, but you’ll also learn some key underpinnings of programming as well. It’s 
+also an object oriented programming (OOP) language. Learning OOP requires students to master a certain level of logic. You have to learn how to think through processes in both
+the specific and the abstract. Again, that provides a good base for learning programming in general.
+
+
+## Tools You Will Need
+
+For setting up environment for performing java codes you need with a minimum of 64 MB of RAM .
+
+You will also need the following softwares −
+- Linux 7.1 or Windows xp/7/8 operating system
+- Java JDK 8
+- Text Editors
+
+
+## Setting up Java development environment
+
+Java SE is freely available from the link [Download Java](https://www.oracle.com/java/technologies/javase/javase-jdk8-downloads.html). You can download a version based on your 
+operating system.Follow the instructions to download Java and run the .exe to install Java on your machine. Once you installed Java on your machine, you will need to set 
+environment variables to point to correct installation directories −
+
+**➤ Setting Up the Path for Windows :-**
+
+Assuming you have installed Java in *c:\Program Files\java\jdk* directory −
+
+- Right-click on 'My Computer' and select 'Properties'.
+- Click the 'Environment variables' button under the 'Advanced' tab.
+- Now, alter the 'Path' variable so that it also contains the path to the Java executable. Example, if the path is currently set to *'C:\WINDOWS\SYSTEM32'*, then change your 
+path to read *'C:\WINDOWS\SYSTEM32;c:\Program Files\java\jdk\bin'*
+
+**➤ Setting Up the Path for Linux, UNIX, Solaris, FreeBSD :-**
+
+Environment variable PATH should be set to point to where the Java binaries have been installed. Refer to your shell documentation, if you have trouble doing this.Example, if
+you use bash as your shell, then you would add the following line to the end of your *'.bashrc: export PATH = /path/to/java:$PATH'*
+
+
+## Popular Java Editors
+
+To write your Java programs, you will need a text editor, you can consider one of the following −
```

Comment:
Maybe instead you can add as IDE instead of text editor

Reply:
should I make another section for IDEs to add Vscode, Vim etc ??
I am confused😕

Reply:
@KhafiaAyyub: Nope you can see, I request the changes [below](https://github.com/devstrons/hello-world/pull/51#discussion_r703707598), hope that makes sense! :slightly_smiling_face: